### DISCUSSION TITLE
How to reset a file in a branch to state  before the branch was created?

### DISCUSSION THREAD

**Discussion by shanemcandrewai** [community/community#23275] [New to GitHub]
<p>Hello,</p>
<p>I have done the following -</p>
<ol>
<li>
<p>Forked a project - <a href="https://github.com/shanemcandrewai/noise_resilient_3dtface/tree/master/code" target="_blank" rel="nofollow noopener">https://github.com/shanemcandrewai/noise_resilient_3dtface/tree/master/code</a></p>
</li>
<li>
<p>Pushed a change to the file <a href="https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/master/readme.md" target="_blank" rel="nofollow noopener">https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/master/readme.md</a></p>
</li>
<li>
<p>Created a new branch fix-typo-generate and pushed a change to another file <a href="https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/fix-typo-generate/code/generate.py" target="_blank" rel="nofollow noopener">https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/fix-typo-generate/code/generate.py</a></p>
</li>
</ol>
<p>Now I want to reset the file <a href="https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/fix-typo-generate/readme.md" target="_blank" rel="nofollow noopener">https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/fix-typo-generate/readme.md</a> to state when I forked the project but only in the fix-typo-generate branch</p>
<p>Investigation -</p>
<p>Display relevant part of the log -</p>
<p>$ git log --oneline --decorate<br>
403c1f8 (HEAD -&gt; fix-typo-generate, origin/fix-typo-generate) fix typo in generate.py --template_shape help<br>
70e8b2e (origin/master, origin/HEAD, master) Added project description<br>
07aa67e Initial commit<br>
….</p>
<p>70e8b2e contains my change to <a href="https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/master/readme.md" target="_blank" rel="nofollow noopener">https://github.com/shanemcandrewai/noise_resilient_3dtface/blob/master/readme.md</a></p>
<p>07aa67e is the version when the project was initially forked</p>
<p>Confirm HEAD points to fix-typo-generate -</p>
<p>$ git branch<br>
* fix-typo-generate<br>
master</p>
<p>prepare to work on commit 07aa67e (version when the project was initially forked) -</p>
<p>$ git checkout 07aa67e<br>
Note: switching to ‘07aa67e’.</p>
<p>You are in ‘detached HEAD’ state. You can look around, make experimental<br>
changes and commit them, and you can discard any commits you make in this<br>
state without impacting any branches by switching back to a branch.</p>
<p>If you want to create a new branch to retain commits you create, you may<br>
do so (now or later) by using -c with the switch command. Example:</p>
<p>git switch -c &lt;new-branch-name&gt;</p>
<p>Or undo this operation with:</p>
<p>git switch -</p>
<p>Turn off this advice by setting config variable advice.detachedHead to false</p>
<p>HEAD is now at 07aa67e Initial commit</p>
<p>Switch back to branch fix-typo-generate -</p>
<p>$ git checkout fix-typo-generate<br>
Previous HEAD position was 07aa67e Initial commit<br>
Switched to branch ‘fix-typo-generate’</p>
<p>$ git show-branch -a fix-typo-generate <br>
* [fix-typo-generate] fix typo in generate.py --template_shape help<br>
! [master] Added project description<br>
! [origin/HEAD] Added project description<br>
! [origin/fix-typo-generate] fix typo in generate.py --template_shape help<br>
! [origin/master] Added project description<br>
-----<br>
* + [fix-typo-generate] fix typo in generate.py --template_shape help<br>
*++++ [master] Added project description</p>
<p>It appears fix-typo-generate in longer version 07aa67e</p>
<p>How can I proceed?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The easiest way would be to create a new branch starting at the commit from which you forked (that’d be 07aa67e, as I understand your description), and then git cherry-pick the commit from the fix-typo-generate branch. Git cherry-pick takes an existing commit and applies it to the current branch. Same diff and commit message, but otherwise an independent commit (there are options to edit the commit message before committing, or to add a line that says where the commit was cherry-picked from).</p>
<p>So:</p>
<ol>
<li>git checkout -b my-new-branch 07aa67e</li>
<li>git cherry-pick 403c1f8</li>
</ol>
